commit c0f9a09adc208fc223ac8e33127c05303633f43c
Author: r0hitsharma <rs@rohit.sh>
Date:   Sun Feb 2 17:31:14 2025 +0100

    Update room status.

diff --git a/app-next/components/Video.jsx b/app-next/components/Video.jsx
index 3eb7a8d..349153c 100644
--- a/app-next/components/Video.jsx
+++ b/app-next/components/Video.jsx
@@ -6,33 +6,35 @@ import { getRoom, updateRoom } from "@/roomActions";
 
 export function Video ({ id, url, cursor, state }) {
   const playedSeconds = useRef(cursor);
-  const [videoState, setVideoState] = useState(state);
+  const videoState = useRef(state);
 
   async function refreshRoom(){
     const room = await getRoom(id);
     playedSeconds.current = room.cursor;
+    videoState.current = room.state;
   }
 
   useEffect(() => {
-    refreshRoom();
+    const intervalId = setInterval(refreshRoom, 3000);
+    return () => clearInterval(intervalId);
   }, []);
 
   function saveRoom() {
-    updateRoom({ id, cursor: playedSeconds.current, video_state: videoState });
+    updateRoom({ id, cursor: playedSeconds.current, video_state: videoState.current });
   }
 
   return <ReactPlayer url={url} width={"100%"} height={"100%"} onProgress={x => {
     playedSeconds.current = parseInt(x.playedSeconds);
     saveRoom();
   }} controls
-  playing={videoState === 'playing'}
-  muted={videoState === 'playing'}
+  playing={videoState.current === 'playing'}
+  muted={videoState.current === 'playing'}
   onPlay={() => {
-    setVideoState('playing');
+    videoState.current = 'playing';
     saveRoom();
   }}
   onPause={() => {
-    setVideoState('paused');
+    videoState.current = 'paused';
     saveRoom();
   }}
   config={{
